src/blk_NLP.jl

```diff
@@ -0,0 +1,13 @@
+abstract type BlockNLP end
+mutable struct BlockNLPModel{T} <: BlockNLP
```

Comment:
Emphasize that `T` is a NLPModel here. As:
```julia
BlockNLPModel{T<:AbstractNLPModel}
```

Reply:
Done.